README.md

```diff
@@ -57,8 +57,12 @@ yunohost app setting synapse turnserver_tls_port
 yunohost app setting synapse turnserver_alt_tls_port
 
 ```
+The turnserver will also chose a port dynamicly when a new call start. The is between 49152 - 49192.
+
 To have a fully functional turnserver you need to open these ports (if it is not automatically done) on your ISP box.
 
+To prevent the situation when the server is behind a NAT, the public IP is written in the turnserver config. By this the turnserver can send his real public IP to the client. For more information see [the coturn example config file](https://github.com/coturn/coturn/blob/master/examples/etc/turnserver.conf#L102-L120). So if you IP change the turnserver config might be obsolete. To fix that you just need to upgrade the app manually. By this the config will be automatically fixed.
```

Comment:
> So if you IP change the turnserver config might be obsolete. To fix that you just need to upgrade the app manually.

Really ?
Don't you think that it's not a good solution ?

Reply:
If you really need to get the public IP, you should add a cron task that check regularly the IP, and update it if needed.

Reply:
I thought also about this solution. So how long should be this cron (10min maybe) ?

In  this case, we need to add a new parameter to ask the user if he has an IP which change sometime (because I'm not for the idea to impose to all user a cron for that). This make a big question about how could we set this value for the old install ? (We might have the same question about some other settings, for an other PR that I'm working on actually).

Reply:
Actually, I don't understand why coturn need to know exactly the public IP.
I mean, usually, when a service talk to another one, it's doing that by sending its IP.

And, I think that most of our users are using dynamic IP, since it's quite seldom to have a static IP.

Reply:
It's explained here : https://github.com/coturn/coturn/blob/master/examples/etc/turnserver.conf#L102-L125

Because, if coturn is behind a NAT, he can't know his public IP. And yes it could be a good question : why coturn need to know his public IP ? I think it's because it need to inform the IP address of the relay. I'm not a specialist of this but after the time that I spent on this I could try to explain how coturn will run : 

- A new call start
- The both client will ask coturn to know the IP of the remote client. (This is the STUN protocol). After the both client will try to establish a connexion between the both client throught the NAT between the client.
- If this fail, the client try to use the TURN protocol witch use the server as a relay. In this case the client contact coturn to know where he could send the call stream.
- Coturn will probably use one port in the port range per client to receive the call stream from each client. Coturn will sent to the client the port and the external IP  (it in this case that coturn might use the `external-ip` option) to the client.

Hope that it will help you to understand the problem.

Reply:
So if I get it correctly, when you write "The turnserver don't work really well.", you mean that calls are failing because direct connection failed (two first bullets) and the current configuration doesn't allow the turn server to relay calls (two last bullets)...?

Reply:
Not really. In the last config coturn was configured only as a turn server (as a relay). The direct connection (stun) was disabled in the config.

Reply:
So how come it worked at least a little without that ports open...?

Reply:
It was probably broken